app/controllers/v1/projects_controller.rb

```diff
@@ -76,6 +76,30 @@ def destroy
       end
     end
 
+    swagger_api :list_tasks do
+      summary "List all of a project's tasks"
+      notes "This lists all of the selected project's the active tasks"
+      param :id
+    end
+    def list_tasks
+      project = Project.find_by(id: params[:id])
+      tasks = project.tasks
+      render json: tasks, each_serializer: V1::TaskSerializer
+    end
+
+    swagger_api :create do
+      summary 'Creates a new Task for the selected Project'
+      param :name, :project_id
+    end
+    def create_task
+      task = Task.new(name: params[:name], project_id: params[:project_id])
```

Comment:
Having the description be optional is totally cool and makes a lot of sense.
The problem is what you have here is:
`task = Task.new(name: params[:name], project_id: params[:project_id])`

You are choosing to explicitly list of the params to build up the task and are not including the description. So even if they sent it, it wouldn't be set.

There are a few ways you can handle this to achieve what you want:

1. Most simple would be to just do `task = Task.new(params)`. That would work, but it's not recommended as you might not want to allow them to send in anything. For example they could send in their own `created_at` with this method.
2. Next would be to do what you are doing but include all the attributes you want to allow being set like `task = Task.new(name: params[:name], description: params[:description], project_id: params[:project_id])`. Setting all the attributes like this does not mean they are all required. If they are missing they'll just be set to `nil`. This is a reasonable method, but doesn't allow you to require at the controller level that something is actually there. That's ok though because if you have appropriate validations on your model then there will be errors in the save.
3. An alternative to number 2 would be to use StrongParameters which Rails gives you to describe what is allowed or not allowed in the input. This allows you to do things like say what is required as well as what is allowed. It might look something like:
```
def create_task
  task = Task.new(task_params)
  if task.save
    ...
  else
    ...
  end
end

private

def task_params
  # You can also require, specify nesting, and more, see docs for info
  params.permit(:project_id, :name, :description)
end
```

Finally, when doing something like creating a dependent object (i.e. Task needs a Project in order to exist). It's often a good idea to pull the Project from the database before creating the Task. You can either use a standard `find` and a 404 will be thrown if it isn't there, or you can use a `find_by` if you want to do your own custom error handling.